containers/proxy-squid-image/configure_and_run.py

```diff
@@ -0,0 +1,205 @@
+import re
+import subprocess
+import sys
+import yaml
+
+
+class UyuniConfigReader:
+    def __init__(self, config_file_path="/etc/uyuni/config.yaml"):
+        self.config_file_path = config_file_path
+
+    def read_config(self):
+        with open(self.config_file_path) as source:
+            return yaml.safe_load(source)
+
+
+class SquidConfigReader:
+    def __init__(self, squid_conf_path="/etc/squid/squid.conf"):
+        self.squid_conf_path = squid_conf_path
+
+    def read_config(self):
+        with open(self.squid_conf_path, "r") as f:
+            return f.read()
+
+    def write_config(self, squid_config):
+        with open(self.squid_conf_path, "w") as f:
+            f.seek(0, 0)
+            f.write(squid_config)
+            f.truncate()
+
+
+class SquidConfigurator:
```

Comment:
The config script is getting more and more complex to I've decided to rewrite it in a slightly more OOP way and added some simple unittests to make it a bit more transparent